### PR TITLE
feat: instrument http gem

### PR DESCRIPTION
**Pull request by robertlaurin**
Provides instrumentation for https://github.com/httprb/http

### CODE REVIEW

File: instrumentation/http/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# OpenTelemetry Http Instrumentation
+
+Todo: Add a description.
```
Comment (fbogsany) [inline instrumentation/http/README.md:3]: 🤔

File: instrumentation/http/lib/opentelemetry/instrumentation/http/patches/client.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Http
+      module Patches
+        # Module to prepend to HTTPClient for instrumentation
```
Comment (fbogsany) [inline instrumentation/http/lib/opentelemetry/instrumentation/http/patches/client.rb:11]:
```suggestion
        # Module to prepend to HTTP::Client for instrumentation
```

File: instrumentation/http/lib/opentelemetry/instrumentation/http/patches/client.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Http
```
Comment (fbogsany) [inline instrumentation/http/lib/opentelemetry/instrumentation/http/patches/client.rb:9]: I know this is all auto-generated, but should this be:
```suggestion
    module HTTP
```

File: instrumentation/http/lib/opentelemetry/instrumentation/http/patches/client.rb (hunk 2 above)
Comment (robertlaurin) [reply instrumentation/http/lib/opentelemetry/instrumentation/http/patches/client.rb:11]: 🙈

**Comment by fbogsany** [comment]
Some small things, but this looks great, thanks!

**Review by fbogsany — APPROVED** [review]

**Comment by robertlaurin** [comment]
@fbogsany https://github.com/open-telemetry/opentelemetry-ruby/pull/618

**Pull request merged**